pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -1460,6 +1493,30 @@ func newUnschedulablePods(unschedulableRecorder, gatedRecorder metrics.MetricRec
 	}
 }
 
+type NominatedPod struct {
+	Name      string
+	Namespace string
+	UID       types.UID
+}
+
+func PodToNominatedPod(pod *v1.Pod) NominatedPod {
```

Comment:
I don't quite understand why we need to use `NominatedPod` instead of `*v1.Pod`? Just to reduce memory usage? Does it need to be a pointer?

Reply:
Maybe rename to `NominatedPodRef`?